boskos/client/client.go

```diff
@@ -640,3 +693,34 @@ func isDialErrorRetriable(err error) bool {
 	}
 	return false
 }
+
+// workFunc describes retrieable work. It should
+// * Return an error for non-recoverable errors
+// * Write retriable errors into `retriedErrs` and return with false, nil
+// * Return with true, nil on success
+type workFunc func(retriedErrs *[]error) (bool, error)
+
+// SleepFunc is called when requests are retried. This may be replaced in tests.
+var SleepFunc = time.Sleep
+
+func retry(work workFunc) error {
+	var retriedErrs []error
+
+	maxAttempts := 4
+	for i := 0; i < maxAttempts; i++ {
+		success, err := work(&retriedErrs)
+		if err != nil {
+			return err
+		}
+		if success {
+			return nil
+		}
+		if i == maxAttempts-1 {
+			break
+		}
+
+		SleepFunc(time.Duration(math.Pow(float64(i+1), 2)) * time.Second)
```

Comment:
This can be just `(i+1) * (i+1) * time.Second` (or `i * i * …`, see the first comment), `pow` is a heavy operation.